### PR TITLE
Storage alert

### LINKED ISSUE #419: [FEATURE] Account storage alert
### Issue to be solved

Background:
https://github.com/Outblock/FRW/issues/61

#### Get Storage and Available Balance Info

Our script API returns `getAccountInfo` cadence, which allow wallet to get the storage info under an account.
```swift
access(all) struct Result {
  access(all) let address: Address
  access(all) let balance: UFix64
  access(all) let availableBalance: UFix64
  access(all) let storageUsed: UInt64
  access(all) let storageCapacity: UInt64
  access(all) let storageFlow: UFix64,
}
```
If `storageCapacity - storageUsed < 10000` then it's insufficient storage

#### Notification
Currently, we use firebase remote config to control the notification messages. It use `news` at the moment.
For insufficient storage alert, we gonna add a new condition flag `insufficientStorage`, which user reach storage limit. 

#### Onchain Interaction
Any onchain Interaction, wallet need do a async check whether the storage is sufficient or not.

If it's send Flow Token or move action ( move fee ), the balance will be affected we need take count of the sending or move fee balance.  ( availableBalance - sent amount ) < 0.001
Other onchain Interaction, we only do the availableBalance check.

Following is the list of storage warning in action button:

1. Move token
2. Move NFT
3. Send Token
4. Send NFT
5. Sign Transaction on browser
6. Sign Transaction on WalletConnect

#### Transaction Listener
Every transaction submitted to the chain, the wallet will listen the callback. So when there is `storageCapacityExceeded`.
we need show an global alert telling user that the last transaction was failing due to storage.

#### Storage Usage Card
Storage info card need to be displayed only under FLOW token detail page.

Figma: 
https://www.figma.com/design/mSr5uIApRJNrpkmN0YNuD6/FLOW-REF-Mobile-Design-v2.0.0?node-id=35558-33202&t=eegpA2haFKcXcypT-1

### PR DESCRIPTION
**Pull request by jaymengxy**
## Related Issue
Closes #419 #425 #426 #450 #467 
## Summary of Changes
Storage Alert

## Need Regression Testing

- [x] Yes
- [ ] No

## Risk Assessment

- [ ] Low
- [ ] Medium
- [x] High

### CODE REVIEW

File: app/src/main/java/com/flowfoundation/wallet/manager/account/AccountInfoManager.kt [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            }
+            logd(TAG, "getAccountInfo response:${String(response?.bytes ?: byteArrayOf())}")
+            return parseAccountInfoResult(String(response?.bytes ?: byteArrayOf()))
+        } catch (e: Throwable) {
+            loge(e)
+            return null
+        }
+    }
+
+    private fun parseAccountInfoResult(json: String): AccountInfo? {
```
Comment (lmcmz) [inline app/src/main/java/com/flowfoundation/wallet/manager/account/AccountInfoManager.kt:171]: This is no the right way to use it

```kotlin
 Cadence.CADENCE_GET_ACCOUNT_INFO.executeCadence {
                arg { address(address) }
}?.decode<AccountInfo>()
```

File: app/src/main/java/com/flowfoundation/wallet/manager/account/model/AccountInfoResponse.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+data class AccountInfo(
+    val address: String,
+    val balance: BigDecimal,
+    val availableBalance: BigDecimal,
+    val storageUsed: Long,
+    val storageCapacity: Long,
+    val storageFlow: BigDecimal
+)
+
+data class AccountInfoInner(
```
Comment (lmcmz) [inline app/src/main/java/com/flowfoundation/wallet/manager/account/model/AccountInfoResponse.kt:29]: No need for this

**Review by lmcmz — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/flowfoundation/wallet/manager/account/AccountInfoManager.kt (hunk 1 above)
Comment (jaymengxy) [reply app/src/main/java/com/flowfoundation/wallet/manager/account/AccountInfoManager.kt:171]: Using decode throws an error.
Serializer for class 'FlowAddress' is not found.
Mark the class as @Serializable or provide the serializer explicitly.

File: app/src/main/java/com/flowfoundation/wallet/manager/account/model/AccountInfoResponse.kt (hunk 2 above)
Comment (jaymengxy) [reply app/src/main/java/com/flowfoundation/wallet/manager/account/model/AccountInfoResponse.kt:29]: I'll remove this after we switch to KMM

**Review by lmcmz — APPROVED** [review]

**Pull request merged**